### PR TITLE
Test architecture armv7hf

### PR DESCRIPTION
**Pull request by jgsogo**
Add a job to CI with arch ``armv7hf``

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ osx: &osx
    language: generic
 matrix:
    include:
+      - <<: *linux
+        env: CONAN_GCC_VERSIONS=4.9 CONAN_DOCKER_IMAGE=lasote/conangcc49 CONAN_ARCHS=armv7hf
```
Comment (uilianries) [inline .travis.yml:14]: the correct docker image is `lasote/conangcc49-armv7hf`

**Review by uilianries — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (jgsogo) [reply .travis.yml:14]: I took it from here: https://github.com/conan-community/conan-openssl/blob/release/1.0.2o/.travis.yml#L28 maybe we need to check a lot of packages :/

**Comment by uilianries** [comment]
It's necessary skip test_package execution because the ABI is not compatible with x86. When cross-building we check the ABI instead to run the test:

```python
def test_arm(self):
        file_ext = "so" if self.options["libpng"].shared else "a"
        lib_path = os.path.join(self.deps_cpp_info["libpng"].libdirs[0], "libpng.%s" % file_ext)
        output = subprocess.check_output(["readelf", "-h", lib_path]).decode()
assert re.search(r"Machine:\s+ARM", output)
```

File: .travis.yml (hunk 1 above)
Comment (uilianries) [reply .travis.yml:14]: maybe CPT is appending the arch name as suffix :thinking:

Comment (uilianries) [reply .travis.yml:14]: Yes, is getting from arch:

https://github.com/conan-io/conan-package-tools/blob/develop/cpt/packager.py#L497

thanks to Lasote!

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by jgsogo** [comment]
Let's see if this new job gets to the test stage. 🐣

**Comment by uilianries** [comment]
All Travis jobs are failing because outdated apt cache.

You need to add `apt-get update` in .travis/install.sh

**Comment by jgsogo** [comment]
The job fails... I'm not sure if we want to walk this path (adding these jobs to CI), it would require a little bit of work on many, many recipes.

I wanted to test it, now, maybe open an issue at bincrafters and get some feedback?

**Comment by uilianries** [comment]
seems libm is not present maybe we need to install sysroot to solve it.

**Comment by uilianries** [comment]
> I wanted to test it, now, maybe open an issue at bincrafters and get some feedback?

Yeah, please open an issue to track this problem :)

**Comment by Croydon** [comment]
Why is this installing apt packages in `.travis/install.sh` instead of inside the recipe?

**Comment by uilianries** [comment]
Good Question :thinking: I updated that file but I didn't think about it

**Comment by theirix** [comment]
At least we should use the `nasm-installer` instead of nasm apt package.

**Comment by uilianries** [comment]
@jgsogo please add the follow code in `build`:

```
if 'arm' in self.settings.arch and self.settings.os == "Linux":
            tools.replace_in_file(os.path.join(self.source_subfolder, 'CMakeListsOriginal.txt'),
                                  'PATHS /usr/lib /usr/local/lib',
                                  'PATHS /usr/lib /usr/local/lib /usr/arm-linux-gnueabihf/lib /usr/arm-linux-gnueabi/lib ')
```

**Comment by jgsogo** [comment]
It works! Great @uilianries!

**Pull request merged**